Framework/Kernel/src/TimeROI.cpp

```diff
@@ -267,37 +267,21 @@ bool TimeROI::isCompletelyInMask(const Types::Core::DateAndTime &startTime,
  * The value is, essentially, whatever it was at the last recorded time before or equal to the one requested.
  */
 bool TimeROI::valueAtTime(const Types::Core::DateAndTime &time) const {
-  if (this->useAll() || time < m_roi.front() || time >= m_roi.back()) {
-    // ignore everything outside of range
+  // ignore any time, if ROI is empty
+  if (this->useAll())
```

Comment:
This can't be right. If the TimeROI is `useAll()` we return that the value is ignore?

Reply:
That is correct, it's on main: [here](https://github.com/mantidproject/mantid/blob/main/Framework/Kernel/src/TimeROI.cpp#L270-L272).  I agree, it's odd, but I kept that aspect as is.  I only re-factored the main use case: time inside the `TimeROI`.

Reply:
Actually, I made a comment there: "ignore any time, if ROI is empty". In other words, if we have no ROI values to choose from, we ignore the time.

Reply:
I have changed that to:
```
 if (useNone() || empty())
    return ROI_IGNORE;
```